mesonbuild/cmake/interpreter.py

```diff
@@ -609,7 +610,7 @@ def _all_lang_stds(self, lang: str) -> T.List[str]:
         for i in res:
             assert isinstance(i, str)
 
-        return res
+        return T.cast(T.List[str], res)
```

Comment:
This `T.cast` looks unnecessary. The previous asserts should already be enough for mypy.

Reply:
I'll have to check, but mypy usually treats unnecessary casts as errors, so it must need it for some reason. Maybe i can just declare the type of the variable?

Reply:
I tested it on my system and mypy doesn't complain if I remove the cast.

`reveal_type(res)` looks a bit interesting though:
```
mesonbuild/cmake/interpreter.py:613:21: note: Revealed type is 'Union[builtins.list[Any], builtins.list[builtins.str]]'
```

Specifically, I don't know why the `List[Any]` is still there after the asserts...

Reply:
yeah, pylance sees the same thing `List | List | List[str]`. I'm not seeing any issues now either, so I'll drop this.